### PR TITLE
[MRG] Improve speed of seasonal tests

### PR DESCRIPTION
**Pull request by tgsmith61591**
* Increase speed of `CHTest` Cython computation
* Add `OCSBTest`, which is orders of magnitude faster and addresses part of #88 
* Make "ocsb" default for `nsdiffs` and `auto_arima`
* Add seasonal test benchmarking notebook

### CODE REVIEW

**Comment by tgsmith61591** [comment]
@charlesdrotar after this you can add your benchmarking feature :)

**Comment by tgsmith61591** [comment]
@charlesdrotar TBH I don't really love the idea of having 25-minute tests running... Rather than benchmarking the known slowest tests, can we benchmark a set that are critical in speed? I.e., anything over a certain time limit will fail

File: pmdarima/tests/test_benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def test_oob_for_issue_29(self, benchmark):
+        benchmark(test_arima.test_oob_for_issue_29)
+
+    def test_oob_sarimax(self, benchmark):
+        benchmark(test_arima.test_oob_sarimax)
+
+    @pytest.mark.parametrize('m', ch_test_parameters)
+    def test_ch_test_long(self, benchmark, m):
+        benchmark(test_seasonality.test_ch_test_long, m=m)
```
Comment (tgsmith61591) [inline pmdarima/tests/test_benchmarks.py]: If I simply run `pytest`, is it going to run this?

Comment (charlesdrotar) [reply pmdarima/tests/test_benchmarks.py]: Yes, unless you tell it to skip benchmark tests.

**Pull request merged**